<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <include file="$(find daniels)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find daniels)/launch/includes/safety_controller.launch.xml"/>
  
  <arg name="odom_frame_id"   default="$(env ROS_NAMESPACE)/odom"/>
  <arg name="base_frame_id"   default="$(env ROS_NAMESPACE)/base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="custom_param_file" default="$(find daniels)/param/dummy.yaml"/>
  <!-- arg name="robotns" default="123456"/-->

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find daniels)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find daniels)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find daniels)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find daniels)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find daniels)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find daniels)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find daniels)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find daniels)/param/navfn_global_planner_params.yaml" command="load" />
    <!-- external params file that could be loaded into the move_base namespace -->
    <rosparam file="$(arg custom_param_file)" command="load" />
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="cmd_vel_mux/input/navi"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
    <remap from="map" to="/map"/>
  </node>
</launch>
